### DISCUSSION TITLE
Android client /kbd param

### DISCUSSION THREAD

**Discussion by drpapp** [FreeRDP/FreeRDP#9091] [General]
hello,

Based on this info [#9087 ](https://github.com/FreeRDP/FreeRDP/pull/9087) I tried to add additional parameters to the kbd param:

 /kbd:[layout:[0x<id>|<name>],lang:<0x<id>>,fn-key:<value>,type:<value>
          subtype:<value>,unicode[:on|off],remap:<key1>=<value1>
          remap:<key2>=<value2>]      Keyboard related options:* layout: set the
                                      keybouard layout announced to the server*
                                      lang: set the keyboard language identifier
                                      sent to the server* fn-key: Function key
                                      value

like  args.add("/kbd:unicode:on, lang:0x0409"), but it is not working.
Tried remap, but also no success.

The problem is that I hit '0', but got hungarian char 'ö', this is why I try to set the lang.

Another problem is that I uncomment some line in the KeyboardMapper file, because the device(barcode reader) has physical keyboard and should enable some additional key like up/down/left/right:

keymapAndroid[KeyEvent.KEYCODE_DPAD_DOWN] = VK_DOWN;
keymapAndroid[KeyEvent.KEYCODE_DPAD_LEFT] = VK_LEFT;
keymapAndroid[KeyEvent.KEYCODE_DPAD_RIGHT] = VK_RIGHT;
keymapAndroid[KeyEvent.KEYCODE_DPAD_UP] = VK_UP;

UP and DOWN started to work, but sending wrong code:
[Java_com_freerdp_freerdpcore_services_LibFreeRDP_freerdp_1send_1key_1event]: send_key_event: 0,

The LEFT/RIGHT buttons not working at all.
Can you please help how to enable them and set the correct send_key value?

Thank in advance.

**Comment by akallabeth** [comment]
if you use unicode input there is no keymap. your local system generates the wrongly mapped key, so check that instead.
